go.mod

```diff
@@ -20,7 +20,7 @@ require (
 	github.com/ipfs/go-log v1.0.4
 	github.com/keep-network/keep-common v1.2.1-0.20201020114759-19c123cbd4f4
 	github.com/keep-network/keep-core v1.3.0
-	github.com/keep-network/tbtc v1.1.1-0.20201020115551-5f9077c74826
+	github.com/keep-network/tbtc v1.1.1-0.20201026093513-cb9246987718
```

Comment:
Nope, we don't have to do anything in `keep-common`.

Reply:
Don't we need generators for eventlog? From the description of the referenced PR:

> Added a TBTCSystemEventLog allowing to access past events emitted by the TBTCSystem contract. This is a temporary workaround because contract bindings generated by the keep-common generator don't support this feature yet.

Reply:
We just need the `abi` file which is already generated and tracked by the VCS.

Reply:
We do need to implement a change to the generator in `keep-common` so that it generates `TBTCSystemEventLog`, correct?

Let me put it this way:

We build tBTC go package from scratch:
- we remove all generated files from ` pkg/chain/ethereum/gen`
- we do `go generate ./...` there
- we push a new version

Then we build `keep-ecdsa`:
- we update a dependency to `tbtc`
- we run `go build`

My understanding is that the compilation will fail because `TBTCSystemEventLog` will not be there.

In other words, we are depending on a file that *should be* generated but it's not generated.

Reply:
Not quite. Eventually, we should extend the existing `keep-common` generator in that way it generates the `Filter*` methods straight into the `TBTCSystem` binding struct. `TBTCSystemEventLog` is just a temporary workaround created manually without using the generator. It is also tracked by VCS similarly as the regular bindings. So, when we build tBTC package from scratch we shouldn't delete anything. We should only run `go generate ./...` to capture possible contract changes.

Reply:
Right. So just to be clear - I am absolutely fine with merging this PR having `TBTCSystemEventLog` created manually just like now but I do not want to release a new version before we have this file generated or having `Filter*` methods generated into `TBTCSystem` struct.

Reply:
I'll leave this discussion open until the proper solution is implemented. I do not want to block this PR, though.